nautobot/extras/datasources/git.py

```diff
@@ -43,13 +43,17 @@
 logger = logging.getLogger("nautobot.datasources.git")
 
 
-def enqueue_pull_git_repository_and_refresh_data(repository, request):
+def enqueue_git_repository_helper(repository, request, func):
     """
-    Convenience wrapper for JobResult.enqueue_job() to enqueue the pull_git_repository_and_refresh_data job.
+    GIt repository enqueue helper function
     """
+    enqueues = {
+        "pull_git_repository_and_refresh_data": pull_git_repository_and_refresh_data,
+        "git_repository_diff_origin_and_local": git_repository_diff_origin_and_local,
+    }
```

Comment:
I don't see a need to do this level of indirection toward the worker functions - why not just pass the worker function directly to `enqueue_git_repository_helper()`, i.e.:

```python
def enqueue_git_repository_diff_origin_and_local(repository, request):
    """Convenience wrapper for JobResult.enqueue_job() to enqueue the git_repository_diff_origin_and_local job."""
    enqueue_git_repository_helper(repository, request, git_repository_diff_origin_and_local)

def enqueue_git_repository_helper(repository, request, func):
    """
    Git repository enqueue helper function
    """
    git_repository_content_type = ContentType.objects.get_for_model(GitRepository)
    JobResult.enqueue_job(
        func,
        repository.name,
        ...
```